### PR TITLE
docs(demos): autocomplete examples

### PR DESCRIPTION
**Pull request by amcdnl**
Adds basic set of demos for autocomplete component:

- Basic
- Filtering
- Display Value
- Better overview example

This is a new PR in follow-up to #5464 merge issues.

### CODE REVIEW

File: src/material-examples/autocomplete-display/autocomplete-display-example.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.example-form {
+  width: 500px;
+}
```
Comment (willshowell) [inline src/material-examples/autocomplete-display/autocomplete-display-example.css]: @amcdnl Think you could add `max-width: 100%` before it merges, here and elsewhere? If not, I'll be happy to in subsequent pr.

https://github.com/angular/material2/pull/5839

Comment (amcdnl) [inline src/material-examples/autocomplete-display/autocomplete-display-example.css]: @willshowell - I was thinking about going through and removing all these fixed widths since it causes responsive issues. I'd like to do that in a separate PR.
